crates/ty_server/src/server/api/notifications/did_change_watched_files.rs

```diff
@@ -107,7 +107,7 @@ impl SyncNotificationHandler for DidChangeWatchedFiles {
                     publish_diagnostics(session, &key, client);
                 }
             }
-
+            publish_settings_diagnostics(session, client);
```

Comment:
We should skip calling `publish_settings_diagnostics` if the client uses workspace diagnostics as it is unnecessary

Reply:
I moved the check inside the call so each callsite doesn't need to think about it.